### DISCUSSION TITLE
About uploading files using filesService.uploadOne

### DISCUSSION THREAD

**Discussion by zhuyifang** [directus/directus#7993] [Archive]
I used some files in the custom api interface
After processing, it is necessary to save the qualified files on the server, but when using this interface, we encountered some problems

I use the following code, but none of the programs work

```js
let readStream = fs.createReadStream(file.path);
const filesService = new FilesService({schema: req.schema})
const aa = await filesService.uploadOne(readStream, {folder: 'f38415ad-2a87-4e68-81a1-2e8fa8c12130'})
```

**Comment by zhuyifang** [comment]
Is there any documentation for interface services such as FilesService?

**Comment by rijkvanzanten** [comment]
There's no documentation around these internal services available at the moment.

"filename_disk" and "storage" are required in that second object:

https://github.com/directus/directus/blob/0fd26d95971bd23a10006b081b0aa2e2e248508b/api/src/services/files.ts#L26-L30

Try using 

```js
let readStream = fs.createReadStream(file.path);
const filesService = new FilesService({schema: req.schema})
const aa = await filesService.uploadOne(readStream, {folder: 'f38415ad-2a87-4e68-81a1-2e8fa8c12130', storage: 'local', filename_disk: 'my-file.png' })
```

(replace "local" with whatever your storage adapter is called)

**Reply by AlexVanK** [reply]
@rijkvanzanten, to wipe a file from the disk, is this correct?
```JS
async function DELETE__file(FilesService, schema, filesUUID) {
   let files = new FilesService({ schema: schema });
   let promise = await files.deleteMany(filesUUID);
   return promise;
}
```
because it doesn't seem to work...
the promise returns just the UUID, but nothing else happens

**Reply by rijkvanzanten** [reply]
That looks correct to me. The promise you're returning are the IDs you (successfully) deleted.

These are the actual methods:

https://github.com/directus/directus/blob/cc076494ff9ddefec97741fda281ef55922451fe/api/src/services/files.ts#L192-L227

**Reply by AlexVanK** [reply]
Yeap, I read that source code too,
however the file is not removed from the server/disk :(

**Reply by rijkvanzanten** [reply]
What storage adapter are you using? 🤔

**Reply by AlexVanK** [reply]
Local filesystem, Windows Server

**Reply by rijkvanzanten** [reply]
@AlexVanK Could you open a new issue with a minimal reproduction we can use to test? That should work (though I can't test, as I'm not a windows user myself)
